components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,463 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent') -- Note: This can be overwritten
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local WidgetInjector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+local Placement = Class.new(function(self, ...) self:init(...) end)
+
+local TODAY = os.date('%Y-%m-%d')
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+	'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
```

Comment:
using numbers to identify columns seems like a really bad idea. impossible to debug

Reply:
if numbers are used for ordering purposes - I suggest putting `index` field in `prizeData` props and computing it with a dedicated mechanism. 

I had to do this recently for a particularly complex distribution table
https://liquipedia.net/commons/Module:EptCountryDist/Starcraft/next
-> uses https://liquipedia.net/commons/Module:DistTable

it's complicated because continent columns contain nested inner columns for countries

Reply:
It has been dropped.